zypp/Fetcher.cc

```diff
@@ -535,59 +503,58 @@ namespace zypp
       }
   }
 
-  void Fetcher::Impl::provideToDest( MediaSetAccess &media, const OnMediaLocation &resource, const Pathname &dest_dir, const Pathname &deltafile )
+  void Fetcher::Impl::provideToDest( MediaSetAccess & media_r, const Pathname & destDir_r , const FetcherJob_Ptr & jobp_r )
   {
-    bool got_from_cache = false;
-
-    // start look in cache
-    got_from_cache = provideFromCache(resource, dest_dir);
+    const OnMediaLocation & resource( jobp_r->location );
 
-    if ( ! got_from_cache )
+    try
     {
-      MIL << "Not found in cache, downloading" << endl;
+      shared_ptr<MediaSetAccess::ReleaseFileGuard> releaseFileGuard; // will take care provided files get released
```

Comment:
scoped_ptr should be enough here ( or std::unique_ptr )

Reply:
Right. The initial guard used a custom deleter, but no longer need it.